test/end2end_test.go

```diff
@@ -7032,6 +7032,168 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
+type testAuthInfo struct {
```

Comment:
Maybe a more descriptive name instead of `testAuthInfo`, and a comment saying how it is used.